ToolBox/AlgoSeekOptionsConverter/AlgoSeekOptionsConverter.cs

```diff
@@ -0,0 +1,164 @@
+﻿/*
+ * QUANTCONNECT.COM - Democratizing Finance, Empowering Individuals.
+ * Lean Algorithmic Trading Engine v2.0. Copyright 2014 QuantConnect Corporation.
+ * 
+ * Licensed under the Apache License, Version 2.0 (the "License"); 
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+using System;
+using System.IO;
+using QuantConnect.Logging;
+using QuantConnect.Data.Market;
+using System.Linq;
+using System.Diagnostics;
+using System.Threading.Tasks;
+
+namespace QuantConnect.ToolBox.AlgoSeekOptionsConverter
+{
+	public class AlgoSeekOptionsConverter
+	{
+		public static void ConvertToFineResolution(string sourceDirectory, string destinationDirectory, long flushInterval)
+		{
+			var quoteProcessor = new InMemoryDataProcessor(TickType.Quote, x => ((Tick) x).TickType == TickType.Quote);
+			var tradeProcessor = new InMemoryDataProcessor(TickType.Trade, x => ((Tick) x).TickType == TickType.Trade);
+
+			var files = Directory.EnumerateFiles(sourceDirectory).OrderByDescending(x => new FileInfo(x).Length);
+			var optionsReaders = files.Select(file => new AlgoSeekOptionsReader(file)).ToList();
+
+			var parsingTime = new Stopwatch();
+			var processingTime = new Stopwatch();
+			var diskTime = new Stopwatch();
+
+			var totalLinesProcessed = 0L;
+			while(optionsReaders.Any(reader => reader.HasNext))
+			{
+				parsingTime.Start();
+				var activeReaders = optionsReaders.Where(reader => reader.HasNext);
+				var nextTick = activeReaders.EarliestTick().Take();
+				var dataTime = nextTick.Time;
+				parsingTime.Stop();
+
+				processingTime.Start();
+				quoteProcessor.Process(nextTick);
+				tradeProcessor.Process(nextTick);
+				processingTime.Stop();
+
+				totalLinesProcessed++;
+				if (totalLinesProcessed % flushInterval == 0)
+				{
+					var cutoffTime = new DateTime(dataTime.Year, dataTime.Month, dataTime.Day, dataTime.Hour, dataTime.Minute, 0, 0) - TimeSpan.FromMinutes(1);
+
+					Log.Trace("Processed {0,3}M lines; Data Time is: {1}; Memory in use: {2} MB", totalLinesProcessed/flushInterval, dataTime, Info.CurrentMemoryUsage);
```

Comment:
In general we don't like statics like Info in Lean, please just make it a property of the class or private variable of the class.